### PR TITLE
127 bug validating a nextjs api endpoint

### PR DESCRIPTION
**Pull request by leewan-09**
### Changes proposed

- Validated create user data
- Validated create project data
- Added Path Aliases for (`@/schema/`, `@/lib/`, `@/components/`)

### Screenshots
**Validate Create user data**

Screenshot 2022-09-13 at 21 59 24

**Validate Create post data**

Screenshot 2022-09-13 at 21 58 38

### CODE REVIEW

**Comment by rohitdasu** [comment]
thanks @leewan-09 will review it 👍

**Comment by rohitdasu** [comment]
I very much liked the path aliases ❤️

**Review by rohitdasu — APPROVED** [review]
just a small request, could you please add index.ts file inside `/schema` folder? and export every schemas from there? it would be easier to import. but i'm approving it.  hope you'll do it.

**Review by RakeshSangem — APPROVED** [review]

File: pages/api/user/index.ts [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-import { prisma } from '../../../lib/prisma';
+import { prisma } from '@/lib/prisma';
 import { User } from '@prisma/client';
+import bodyValidator from '@/lib/bodyValidator';
+import {
+  errorResponse,
+  successResponse,
+  validationResponse,
+} from '@/lib/http.response';
+import { userSchema } from '@/schema/userSchema';
```
Comment (rohitdasu) [inline pages/api/user/index.ts:10]:
```suggestion
import { userSchema } from '@/schema';
```

Comment (leewan-09) [reply pages/api/user/index.ts:10]: image

Comment (rohitdasu) [reply pages/api/user/index.ts:10]: could you try @/schema/index?

Comment (leewan-09) [reply pages/api/user/index.ts:10]: this way it does work `import { postSchema } from '@/schema/index';`

Comment (rohitdasu) [reply pages/api/user/index.ts]: then use like this. later we can look at this issue. benefit is we get all schema from one import only.

**Pull request merged**